fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/socialmedia/SocialMediaViewModel.kt

```diff
@@ -287,7 +293,9 @@ constructor(
       val toSearch = token.publicKey.encoded
       senders.contains(toSearch)
     } catch (e: KeyException) {
-      logAndShow(getApplication(), TAG, e, R.string.error_retrieve_own_token)
+      if (_hasValidToken.value == true) {
```

Comment:
Can't you remove the "== true" in the if ?

Reply:
I am not sure I can since the value is `Boolean?` (could be null), but I'll check it out while addressing the other comments :thum

Reply:
In fact, I could not. I tried setting up the value to false in an early stage but did not change it, I guess it's really just a detail